### ISSUE TITLE
After some hours/days fauxmoESP device doesn't react to Alexa anymore

### ISSUE THREAD

**Issue by samtrot** [vintlabs/fauxmoESP#162]
Hello,

I have setup everything in a way that it works with Alexa and I can control my ESP8266 with fauxmoESP. So thank you for the great work1

However, after some hours or even days, Alexa doesn't find the device anymore - "please check your network connection" etc.

Is there any sleep mode or something? Is this a know problem? What could I do to prevent it?

Best Regards

**Comment by sebkouba** [comment]
It could be several things. 

Can you still ping the device? That would indicate wireless issues which would be my first guess. As a workaround you could restart the device when you don't need it (at night?) or implement a wifi reconnection function. I've never had issues with devices disappearing so I'd start looking other directions.

**Comment by pvint** [comment]
I had the same problem with one WiFi network about a year ago (I don't recall what type of router it was). On other networks they seem to stay connected for a long time with the same code.

Devices would run for a day or so, then unreachable. I tried a few things to detect when it had lost connection, but was not able to sort it out.

What I ended up doing was using ESP32-Ping (there's an ESP8266Ping library too, but haven't tested it). I set it up to ping the router once every 15 minutes, and if it could not reach it, disconnect the WiFi, reconnect, and try again.
